test/functional/aggregation_tests.js

```diff
@@ -1,7 +1,167 @@
 'use strict';
 var expect = require('chai').expect;
+const ReadPreference = require('mongodb-core').ReadPreference;
+const sinon = require('sinon');
 
 describe('Aggregation', function() {
+  it('Should use secondary readPreference if given', {
+    metadata: { requires: { topology: 'replicaset' } },
+
+    test: function(done) {
+      const configuration = this.configuration;
+
+      const client = configuration.newClient(
+        { w: 1, readConcern: { level: 'majority' } },
+        { poolSize: 1 }
+      );
+
+      // Trigger test once whole set is up
+      client.connect((err, client) => {
+        expect(err).to.be.null;
+
+        const db = client.db(this.configuration.db);
+        // Some docs for insertion
+        const docs = [
+          {
+            title: 'this is my title',
+            author: 'bob',
+            posted: new Date(),
+            pageViews: 5,
+            tags: ['fun', 'good', 'fun'],
+            other: { foo: 5 },
+            comments: [
+              { author: 'joe', text: 'this is cool' },
+              { author: 'sam', text: 'this is bad' }
+            ]
+          }
+        ];
+
+        // Create a collection
+        const collection = db.collection(
+          'shouldCorrectlyExecuteSimpleAggregationPipelineWithSecondaryReadPreference'
+        );
+        // Insert the docs
+        collection.insert(docs, { w: 1 }, (err, result) => {
+          expect(result).to.exist;
+          expect(err).to.be.null;
+
+          const internalClientCursor = sinon.stub(client.topology.s.coreTopology, 'cursor');
+          const expectedReadPreference = new ReadPreference(ReadPreference.SECONDARY);
+
+          // Execute aggregate, notice the pipeline is expressed as an Array
+          collection.aggregate(
+            [
+              {
+                $project: {
+                  author: 1,
+                  tags: 1
+                }
+              },
+              { $unwind: '$tags' },
+              {
+                $group: {
+                  _id: { tags: '$tags' },
+                  authors: { $addToSet: '$author' }
+                }
+              }
+            ],
+            { readPreference: new ReadPreference(ReadPreference.SECONDARY) },
+            (err, cursor) => {
+              expect(err).to.be.null;
+              expect(cursor).to.not.be.null;
+              expect(internalClientCursor.getCall(0).args[2])
```

Comment:
This confirms that the readPreference is getting to the cursor, but i don't think it confirms that the actual command respects the readPreference. Which are are you trying to test?